src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -268,52 +293,54 @@ protected virtual void AppendData(StringBuilder sb, Exception ex)
             }
         }
 
-        private ExceptionDataTarget[] CompileFormat(string formatSpecifier)
+        /// <summary>
+        /// Split the string and then compile into list of Rendering formats.
+        /// </summary>
+        /// <param name="formatSpecifier"></param>
+        /// <returns></returns>
+        private List<ExceptionRenderingFormat> CompileFormat(string formatSpecifier)
         {
+            List<ExceptionRenderingFormat> formats = new List<ExceptionRenderingFormat>();
             string[] parts = formatSpecifier.Replace(" ", string.Empty).Split(',');
-            var dataTargets = new List<ExceptionDataTarget>();
 
             foreach (string s in parts)
             {
+                ExceptionRenderingFormat renderingFormat;
+
                 switch (s.ToUpper(CultureInfo.InvariantCulture))
```

Comment:
I think it's still nice to rewrite the `swich` to a static dictionary. We have discussed this before I think, but we can write an overload. 

the dictionary is better for the code complexity (arcording to Cyclomatic complexity) and so easier to mainain.